app/src/frontend/map-app.tsx

```diff
@@ -75,6 +78,52 @@ class MapApp extends React.Component<MapAppProps, MapAppState> {
         }
     }
 
+    /**
+     * Fetches building data if a building is selected but no data provided through 
+     * props (from server-side rendering)
+     */
+    async fetchBuildingData() {
+        if(this.props.match.params.building != undefined && this.props.building == undefined) {
+            try {
+                // TODO: simplify API calls, create helpers for fetching data
+                const buildingId = strictParseInt(this.props.match.params.building);
+                let [building, building_uprns, building_like] = await Promise.all([
+                    fetch(`/api/buildings/${buildingId}.json`, {
+                        method: 'GET',
+                        headers: {
+                            'Content-Type': 'application/json'
+                        },
+                        credentials: 'same-origin'
+                    }).then(res => res.json()),
+                    fetch(`/api/buildings/${buildingId}/uprns.json`, {
+                        method: 'GET',
+                        headers: {
+                            'Content-Type': 'application/json'
+                        },
+                        credentials: 'same-origin'
+                    }).then(res => res.json()),
+                    fetch(`/api/buildings/${buildingId}/like.json`, {
+                        method: 'GET',
+                        headers: {
+                            'Content-Type': 'application/json'
+                        }, 
+                        credentials: 'same-origin'
+                    }).then(res => res.json())
+                ]);
```

Comment:
We should probably look into writing some helpers for making JSON API calls (or use a third-party library like Axios) to reduce `fetch`-related repetition across the front-end.

Also, it might be good to create a special endpoint for fetching all relevant data for a building/user pair, to reduce the number of requests to the back-end.